### ISSUE TITLE
update to use 5.1

### ISSUE THREAD

**Issue by tkimnguyen** [collective/training-sandbox#11]
5.1.0 is out, and 5.1.1 is pending.

May depend on https://github.com/plone/heroku-buildpack-plone/issues/25

**Comment by witekdev** [comment]
I just tried to use 5.1.2 using the same methodology that we are using currently in this repo
My app built but then didn't want to run:

```
witek@ubuntu:~/dev/training-sandbox$ heroku logs --app https://testing-512-5.herokuapp.com/
 ▸    The requested API endpoint was not found. Are you using the right HTTP verb (i.e. `GET` vs. `POST`), and did you specify your intended version with the `Accept` header?
witek@ubuntu:~/dev/training-sandbox$ heroku logs --app testing-512-5
2018-07-30T22:53:25.736483+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/cfgparser.py", line 143, in handle_directive
2018-07-30T22:53:25.736529+00:00 app[web.1]: self.handle_import(section, arg)
2018-07-30T22:53:25.736531+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/cfgparser.py", line 149, in handle_import
2018-07-30T22:53:25.736577+00:00 app[web.1]: self.context.importSchemaComponent(pkgname)
2018-07-30T22:53:25.736579+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/loader.py", line 306, in importSchemaComponent
2018-07-30T22:53:25.736643+00:00 app[web.1]: ZConfig.schema.parseComponent(resource, self._loader, schema)
2018-07-30T22:53:25.736645+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 38, in parseComponent
2018-07-30T22:53:25.736679+00:00 app[web.1]: xml.sax.parse(resource.file, parser)
2018-07-30T22:53:25.736681+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/__init__.py", line 33, in parse
2018-07-30T22:53:25.737387+00:00 app[web.1]: parser.parse(source)
2018-07-30T22:53:25.737393+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/expatreader.py", line 110, in parse
2018-07-30T22:53:25.737932+00:00 app[web.1]: xmlreader.IncrementalParser.parse(self, source)
2018-07-30T22:53:25.737936+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
2018-07-30T22:53:25.738034+00:00 app[web.1]: self.feed(buffer)
2018-07-30T22:53:25.738035+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/expatreader.py", line 213, in feed
2018-07-30T22:53:25.738076+00:00 app[web.1]: self._parser.Parse(data, isFinal)
2018-07-30T22:53:25.738079+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/expatreader.py", line 317, in start_element
2018-07-30T22:53:25.738130+00:00 app[web.1]: self._cont_handler.startElement(name, AttributesImpl(attrs))
2018-07-30T22:53:25.738133+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 108, in startElement
2018-07-30T22:53:25.738173+00:00 app[web.1]: getattr(self, "start_" + name)(attrs)
2018-07-30T22:53:25.738175+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 342, in start_sectiontype
2018-07-30T22:53:25.738217+00:00 app[web.1]: keytype, valuetype, datatype = self.get_sect_typeinfo(attrs)
2018-07-30T22:53:25.738220+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 206, in get_sect_typeinfo
2018-07-30T22:53:25.738259+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 199, in get_datatype
2018-07-30T22:53:25.738360+00:00 app[web.1]: t = self.search(name)
2018-07-30T22:53:25.738256+00:00 app[web.1]: datatype = self.get_datatype(attrs, "datatype", "null", base)
2018-07-30T22:53:25.738294+00:00 app[web.1]: return self._registry.get(dtname)
2018-07-30T22:53:25.738297+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/datatypes.py", line 429, in get
2018-07-30T22:53:25.738363+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/datatypes.py", line 454, in search
2018-07-30T22:53:25.738424+00:00 app[web.1]: package = __import__(n, g, g, component)
2018-07-30T22:53:25.738427+00:00 app[web.1]: File "/app/eggs/RelStorage-1.6.3-py2.7.egg/relstorage/config.py", line 18, in <module>
2018-07-30T22:53:25.738446+00:00 app[web.1]: from relstorage.options import Options
2018-07-30T22:53:25.738448+00:00 app[web.1]: File "/app/eggs/RelStorage-1.6.3-py2.7.egg/relstorage/options.py", line 22, in <module>
2018-07-30T22:53:25.738467+00:00 app[web.1]: from ZEO.zrpc.connection import Connection as __Connection
2018-07-30T22:53:25.738483+00:00 app[web.1]: ImportError: No module named zrpc.connection
2018-07-30T22:53:26.195559+00:00 heroku[web.1]: State changed from starting to crashed
2018-07-30T22:53:26.197833+00:00 heroku[web.1]: State changed from crashed to starting
2018-07-30T22:53:47.720615+00:00 heroku[web.1]: Starting process with command `python configure_zopeconf.py; bin/instance console`
2018-07-30T22:53:53.821300+00:00 heroku[web.1]: State changed from starting to crashed
2018-07-30T22:53:53.670556+00:00 app[web.1]: Traceback (most recent call last):
2018-07-30T22:53:53.670572+00:00 app[web.1]: File "bin/instance", line 315, in <module>
2018-07-30T22:53:53.670649+00:00 app[web.1]: File "/app/eggs/plone.recipe.zope2instance-4.3-py2.7.egg/plone/recipe/zope2instance/ctl.py", line 610, in main
2018-07-30T22:53:53.670644+00:00 app[web.1]: + sys.argv[1:]))
2018-07-30T22:53:53.670793+00:00 app[web.1]: options.realize(args, doc=__doc__)
2018-07-30T22:53:53.670799+00:00 app[web.1]: File "/app/eggs/Zope2-2.13.27-py2.7.egg/Zope2/Startup/zopectl.py", line 162, in realize
2018-07-30T22:53:53.809851+00:00 heroku[web.1]: Process exited with status 1
2018-07-30T22:53:53.670862+00:00 app[web.1]: ZDOptions.realize(self, *args, **kw)
2018-07-30T22:53:53.670867+00:00 app[web.1]: File "/app/eggs/zdaemon-4.2.0-py2.7.egg/zdaemon/zdoptions.py", line 296, in realize
2018-07-30T22:53:53.670950+00:00 app[web.1]: self.load_configfile()
2018-07-30T22:53:53.670955+00:00 app[web.1]: File "/app/eggs/zdaemon-4.2.0-py2.7.egg/zdaemon/zdoptions.py", line 347, in load_configfile
2018-07-30T22:53:53.671059+00:00 app[web.1]: self.zconfig_options)
2018-07-30T22:53:53.671064+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/loader.py", line 55, in loadConfig
2018-07-30T22:53:53.671077+00:00 app[web.1]: return _get_config_loader(schema, overrides).loadURL(url)
2018-07-30T22:53:53.671100+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/loader.py", line 83, in loadURL
2018-07-30T22:53:53.671133+00:00 app[web.1]: return self.loadResource(r)
2018-07-30T22:53:53.671138+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/loader.py", line 271, in loadResource
2018-07-30T22:53:53.671249+00:00 app[web.1]: self._parse_resource(sm, resource)
2018-07-30T22:53:53.671252+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/loader.py", line 322, in _parse_resource
2018-07-30T22:53:53.671313+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/cfgparser.py", line 66, in parse
2018-07-30T22:53:53.671307+00:00 app[web.1]: parser.parse(matcher)
2018-07-30T22:53:53.671348+00:00 app[web.1]: self.handle_directive(section, line[1:])
2018-07-30T22:53:53.671354+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/cfgparser.py", line 143, in handle_directive
2018-07-30T22:53:53.671419+00:00 app[web.1]: self.handle_import(section, arg)
2018-07-30T22:53:53.671421+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/cfgparser.py", line 149, in handle_import
2018-07-30T22:53:53.671457+00:00 app[web.1]: self.context.importSchemaComponent(pkgname)
2018-07-30T22:53:53.671463+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/loader.py", line 306, in importSchemaComponent
2018-07-30T22:53:53.671538+00:00 app[web.1]: ZConfig.schema.parseComponent(resource, self._loader, schema)
2018-07-30T22:53:53.671544+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 38, in parseComponent
2018-07-30T22:53:53.671591+00:00 app[web.1]: xml.sax.parse(resource.file, parser)
2018-07-30T22:53:53.671594+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/__init__.py", line 33, in parse
2018-07-30T22:53:53.671696+00:00 app[web.1]: parser.parse(source)
2018-07-30T22:53:53.671701+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/expatreader.py", line 110, in parse
2018-07-30T22:53:53.672909+00:00 app[web.1]: xmlreader.IncrementalParser.parse(self, source)
2018-07-30T22:53:53.672914+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/xmlreader.py", line 123, in parse
2018-07-30T22:53:53.673603+00:00 app[web.1]: self.feed(buffer)
2018-07-30T22:53:53.673629+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/expatreader.py", line 213, in feed
2018-07-30T22:53:53.673745+00:00 app[web.1]: self._parser.Parse(data, isFinal)
2018-07-30T22:53:53.673769+00:00 app[web.1]: File "/usr/lib/python2.7/xml/sax/expatreader.py", line 317, in start_element
2018-07-30T22:53:53.673898+00:00 app[web.1]: self._cont_handler.startElement(name, AttributesImpl(attrs))
2018-07-30T22:53:53.673985+00:00 app[web.1]: getattr(self, "start_" + name)(attrs)
2018-07-30T22:53:53.673922+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 108, in startElement
2018-07-30T22:53:53.674009+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 342, in start_sectiontype
2018-07-30T22:53:53.674143+00:00 app[web.1]: keytype, valuetype, datatype = self.get_sect_typeinfo(attrs)
2018-07-30T22:53:53.674166+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 206, in get_sect_typeinfo
2018-07-30T22:53:53.674275+00:00 app[web.1]: datatype = self.get_datatype(attrs, "datatype", "null", base)
2018-07-30T22:53:53.674299+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/schema.py", line 199, in get_datatype
2018-07-30T22:53:53.674403+00:00 app[web.1]: return self._registry.get(dtname)
2018-07-30T22:53:53.674428+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/datatypes.py", line 429, in get
2018-07-30T22:53:53.674574+00:00 app[web.1]: t = self.search(name)
2018-07-30T22:53:53.674599+00:00 app[web.1]: File "/app/eggs/ZConfig-3.1.0-py2.7.egg/ZConfig/datatypes.py", line 454, in search
2018-07-30T22:53:53.674745+00:00 app[web.1]: package = __import__(n, g, g, component)
2018-07-30T22:53:53.674769+00:00 app[web.1]: File "/app/eggs/RelStorage-1.6.3-py2.7.egg/relstorage/config.py", line 18, in <module>
2018-07-30T22:53:53.674819+00:00 app[web.1]: from relstorage.options import Options
2018-07-30T22:53:53.674843+00:00 app[web.1]: File "/app/eggs/RelStorage-1.6.3-py2.7.egg/relstorage/options.py", line 22, in <module>
2018-07-30T22:53:53.674925+00:00 app[web.1]: from ZEO.zrpc.connection import Connection as __Connection
2018-07-30T22:53:53.674998+00:00 app[web.1]: ImportError: No module named zrpc.connection
2018-07-30T22:53:54.544652+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=testing-512-5.herokuapp.com request_id=fb560327-66fa-4db8-b068-77c45c7ca3a8 fwd="206.190.145.84" dyno= connect= service= status=503 bytes= protocol=https
2018-07-30T22:53:55.412699+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=testing-512-5.herokuapp.com request_id=6f44f5a4-97ad-4c6c-859c-263876311310 fwd="206.190.145.84" dyno= connect= service= status=503 bytes= protocol=https
2018-07-30T22:54:09.512763+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=testing-512-5.herokuapp.com request_id=99347951-0dec-49fd-aeb2-34d122ee6d56 fwd="206.190.145.84" dyno= connect= service= status=503 bytes= protocol=https
2018-07-30T22:54:10.110060+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=testing-512-5.herokuapp.com request_id=c8736a7c-0518-4cfe-bb40-ae6f92ffd154 fwd="206.190.145.84" dyno= connect= service= status=503 bytes= protocol=https
```

Perhaps we are best off investigating and using the new method mentioned used by @zupo here:
https://github.com/plone/heroku-buildpack-plone/issues/25

**Comment by witekdev** [comment]
Looks like pinning RelStorage =2.1.0 fixes things for 5.1.2
https://community.plone.org/t/which-good-version-combination-of-relstorage-for-plone-5-x/5249/2

**Comment by witekdev** [comment]
Fixed
